aiobotocore/httpxsession.py

```diff
@@ -0,0 +1,305 @@
+            )
+            # auth, follow_redirects
+            response = await self._session.send(httpx_request, stream=True)
+            response_headers = botocore.compat.HTTPHeaders.from_pairs(
+                response.headers.items()
+            )
+
+            http_response = aiobotocore.awsrequest.HttpxAWSResponse(
+                str(response.url),
+                response.status_code,
+                response_headers,
+                response,
+            )
+
+            if not request.stream_output:
+                # Cause the raw stream to be exhausted immediately. We do it
+                # this way instead of using preload_content because
+                # preload_content will never buffer chunked responses
+                await http_response.content
+
+            return http_response
+
+        # **previous exception mapping**
+        # aiohttp.ClientSSLError -> SSLError
+
+        # aiohttp.ClientProxyConnectiorError
+        # aiohttp.ClientHttpProxyError -> ProxyConnectionError
+
+        # aiohttp.ServerDisconnectedError
+        # aiohttp.ClientPayloadError
+        # aiohttp.http_exceptions.BadStatusLine -> ConnectionClosedError
+
+        # aiohttp.ServerTimeoutError -> ConnectTimeoutError|ReadTimeoutError
+
+        # aiohttp.ClientConnectorError
+        # aiohttp.ClientConnectionError
+        # socket.gaierror -> EndpointConnectionError
+
+        # asyncio.TimeoutError -> ReadTimeoutError
+
+        # **possible httpx exception mapping**
+        # httpx.CookieConflict
+        # httpx.HTTPError
+        # * httpx.HTTPStatusError
+        # * httpx.RequestError
+        #   * httpx.DecodingError
+        #   * httpx.TooManyRedirects
+        # * httpx.TransportError
+        #   * httpx.NetworkError
+        #     * httpx.CloseError -> ConnectionClosedError
+        #     * httpx.ConnectError -> EndpointConnectionError
+        #     * httpx.ReadError
+        #     * httpx.WriteError
+        #   * httpx.ProtocolError
+        #     * httpx.LocalProtocolError -> SSLError??
+        #     * httpx.RemoteProtocolError
+        #   * httpx.ProxyError -> ProxyConnectionError
+        #   * httpx.TimeoutException
+        #     * httpx.ConnectTimeout -> ConnectTimeoutError
+        #     * httpx.PoolTimeout
+        #     * httpx.ReadTimeout -> ReadTimeoutError
+        #     * httpx.WriteTimeout
+        #   * httpx.UnsupportedProtocol
+        # * httpx.InvalidURL
+
+        except httpx.ConnectError as e:
+            raise EndpointConnectionError(endpoint_url=request.url, error=e)
+        except (socket.gaierror,) as e:
+            raise EndpointConnectionError(endpoint_url=request.url, error=e)
+        except asyncio.TimeoutError as e:
+            raise ReadTimeoutError(endpoint_url=request.url, error=e)
+        except httpx.ReadTimeout as e:
+            raise ReadTimeoutError(endpoint_url=request.url, error=e)
+        except NotImplementedError:
+            raise
```

Comment:
please add cmt on raise line that this is to avoid transforming `NotImplementedError` to `HTTPClientError`